orttraining/orttraining/python/training/orttrainer.py

```diff
@@ -886,7 +886,7 @@ def _training_session_run_helper(self, is_train, inputs, inputs_desc, outputs_de
                 test_pt_device = torch.device(target_device)
             except:
                 #in this case, input/output must on CPU
-                assert(input.device == 'cpu')
+                assert(input.device.type == 'cpu')
```

Comment:
it would be nice to add a unit test here, but I'm not sure if that would be feasible since this is an internal api and the CI does not test non-CUDA devices

Reply:
unfortunately we don't have public non-cuda devices... maybe we can solve it later.